pkg/k8s/service_cache.go

```diff
@@ -162,16 +217,36 @@ func (s *ServiceCache) DeleteService(k8sSvc *types.Service) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
-	oldService, serviceOK := s.services[svcID]
-	endpoints, _ := s.correlateEndpoints(svcID)
+	oldSvc, oldSvcOk := s.services[svcID]
+	oldEndpoints, _ := s.correlateEndpoints(svcID)
 	delete(s.services, svcID)
+	if svcID.k8sExternal {
+		// We will delete the external endpoints created in the service addition
+		delete(s.endpoints, svcID)
+
+		svcID.k8sExternal = false
```

Comment:
Related to my comment above - having to manipulate this value to false vs. true in order to run `correlateEndpoints` is not good. Why not change `correlateEndpoints` to accept a parameter which you can use to signal whether the operation should consider external vs. non-external?

Reply:
Because we store the external IPs endpoints in the endpoint map with the key set for a service ID that has external IPs. The service can have external IPs *and* "normal" k8s endpoints. In order to keep track which endpoints belong to each type is easier to define as this in the endpoints map.